samples/Dantooine/Dantooine.Server/Worker.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Dantooine.Server.Data;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using OpenIddict.Abstractions;
+using static OpenIddict.Abstractions.OpenIddictConstants;
+
+namespace Dantooine.Server
+{
+    public class Worker : IHostedService
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public Worker(IServiceProvider serviceProvider)
+            => _serviceProvider = serviceProvider;
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            using var scope = _serviceProvider.CreateScope();
+
+            var context = scope.ServiceProvider.GetRequiredService<ApplicationDbContext>();
+            await context.Database.EnsureCreatedAsync(cancellationToken);
+
+            await RegisterApplicationsAsync(scope.ServiceProvider);
+            await RegisterScopesAsync(scope.ServiceProvider);
+
+            static async Task RegisterApplicationsAsync(IServiceProvider provider)
+            {
+                var manager = provider.GetRequiredService<IOpenIddictApplicationManager>();
+
+                // API
+                if (await manager.FindByClientIdAsync("resource_server_1") == null)
+                {
+                    var descriptor = new OpenIddictApplicationDescriptor
+                    {
+                        ClientId = "resource_server_1",
+                        ClientSecret = "846B62D0-DEF9-4215-A99D-86E6B8DAB342",
+                        Permissions =
+                        {
+                            Permissions.Endpoints.Introspection
+                        }
+                    };
+
+                    await manager.CreateAsync(descriptor);
+                }
+
+                // Blazor Hosted
+                if (await manager.FindByClientIdAsync("blazorcodeflowpkceclient") is null)
+                {
+                    await manager.CreateAsync(new OpenIddictApplicationDescriptor
+                    {
+                        ClientId = "blazorcodeflowpkceclient",
+                        ConsentType = ConsentTypes.Explicit,
+                        DisplayName = "Blazor code PKCE",
+                        DisplayNames =
+                        {
+                            [CultureInfo.GetCultureInfo("fr-FR")] = "Application cliente MVC"
+                        },
+                        PostLogoutRedirectUris =
+                        {
+                            new Uri("https://localhost:44348/signout-callback-oidc"),
+                            new Uri("https://localhost:5001/signout-callback-oidc")
+                        },
+                        RedirectUris =
+                        {
+                            new Uri("https://localhost:44348/signin-oidc"),
+                            new Uri("https://localhost:5001/signin-oidc")
```

Comment:
Same question.

Reply:
yes, depending how the client is started

Reply:
removed and updated